// @api3/contracts/dist/typechain-types/factories/access/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OwnableCallForwarder__factory = exports.HashRegistry__factory = exports.AccessControlRegistryAdminnedWithManager__factory = exports.AccessControlRegistryAdminned__factory = exports.AccessControlRegistry__factory = exports.interfaces = void 0;
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
exports.interfaces = __importStar(require("./interfaces"));
var AccessControlRegistry__factory_1 = require("./AccessControlRegistry__factory");
Object.defineProperty(exports, "AccessControlRegistry__factory", { enumerable: true, get: function () { return AccessControlRegistry__factory_1.AccessControlRegistry__factory; } });
var AccessControlRegistryAdminned__factory_1 = require("./AccessControlRegistryAdminned__factory");
Object.defineProperty(exports, "AccessControlRegistryAdminned__factory", { enumerable: true, get: function () { return AccessControlRegistryAdminned__factory_1.AccessControlRegistryAdminned__factory; } });
var AccessControlRegistryAdminnedWithManager__factory_1 = require("./AccessControlRegistryAdminnedWithManager__factory");
Object.defineProperty(exports, "AccessControlRegistryAdminnedWithManager__factory", { enumerable: true, get: function () { return AccessControlRegistryAdminnedWithManager__factory_1.AccessControlRegistryAdminnedWithManager__factory; } });
var HashRegistry__factory_1 = require("./HashRegistry__factory");
Object.defineProperty(exports, "HashRegistry__factory", { enumerable: true, get: function () { return HashRegistry__factory_1.HashRegistry__factory; } });
var OwnableCallForwarder__factory_1 = require("./OwnableCallForwarder__factory");
Object.defineProperty(exports, "OwnableCallForwarder__factory", { enumerable: true, get: function () { return OwnableCallForwarder__factory_1.OwnableCallForwarder__factory; } });
