// @api3/contracts/dist/typechain-types/factories/api3-server-v1/proxies/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProxyFactory__factory = exports.DataFeedProxyWithOev__factory = exports.DataFeedProxy__factory = exports.DapiProxyWithOev__factory = exports.DapiProxy__factory = exports.mock = exports.interfaces = void 0;
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
exports.interfaces = __importStar(require("./interfaces"));
exports.mock = __importStar(require("./mock"));
var DapiProxy__factory_1 = require("./DapiProxy__factory");
Object.defineProperty(exports, "DapiProxy__factory", { enumerable: true, get: function () { return DapiProxy__factory_1.DapiProxy__factory; } });
var DapiProxyWithOev__factory_1 = require("./DapiProxyWithOev__factory");
Object.defineProperty(exports, "DapiProxyWithOev__factory", { enumerable: true, get: function () { return DapiProxyWithOev__factory_1.DapiProxyWithOev__factory; } });
var DataFeedProxy__factory_1 = require("./DataFeedProxy__factory");
Object.defineProperty(exports, "DataFeedProxy__factory", { enumerable: true, get: function () { return DataFeedProxy__factory_1.DataFeedProxy__factory; } });
var DataFeedProxyWithOev__factory_1 = require("./DataFeedProxyWithOev__factory");
Object.defineProperty(exports, "DataFeedProxyWithOev__factory", { enumerable: true, get: function () { return DataFeedProxyWithOev__factory_1.DataFeedProxyWithOev__factory; } });
var ProxyFactory__factory_1 = require("./ProxyFactory__factory");
Object.defineProperty(exports, "ProxyFactory__factory", { enumerable: true, get: function () { return ProxyFactory__factory_1.ProxyFactory__factory; } });
