// zksync-ethers/build/typechain/IPaymasterFlowFactory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IPaymasterFlowFactory = void 0;
const ethers_1 = require("ethers");
class IPaymasterFlowFactory {
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.IPaymasterFlowFactory = IPaymasterFlowFactory;
const _abi = [
    {
        inputs: [
            {
                internalType: "address",
                name: "_token",
                type: "address",
            },
            {
                internalType: "uint256",
                name: "_minAllowance",
                type: "uint256",
            },
            {
                internalType: "bytes",
                name: "_innerInput",
                type: "bytes",
            },
        ],
        name: "approvalBased",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bytes",
                name: "input",
                type: "bytes",
            },
        ],
        name: "general",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
];
