// @api3/contracts/dist/typechain-types/factories/access/interfaces/IHashRegistry__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IHashRegistry__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                internalType: "bytes32",
                name: "hashType",
                type: "bytes32",
            },
            {
                indexed: false,
                internalType: "bytes32",
                name: "hashValue",
                type: "bytes32",
            },
            {
                indexed: false,
                internalType: "uint256",
                name: "hashTimestamp",
                type: "uint256",
            },
        ],
        name: "RegisteredHash",
        type: "event",
    },
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                internalType: "bytes32",
                name: "hashType",
                type: "bytes32",
            },
            {
                indexed: false,
                internalType: "bytes32",
                name: "hashValue",
                type: "bytes32",
            },
            {
                indexed: false,
                internalType: "uint256",
                name: "hashTimestamp",
                type: "uint256",
            },
        ],
        name: "SetHash",
        type: "event",
    },
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                internalType: "bytes32",
                name: "hashType",
                type: "bytes32",
            },
            {
                indexed: false,
                internalType: "address[]",
                name: "signers",
                type: "address[]",
            },
        ],
        name: "SetSigners",
        type: "event",
    },
    {
        inputs: [
            {
                internalType: "bytes32",
                name: "hashType",
                type: "bytes32",
            },
        ],
        name: "getHashValue",
        outputs: [
            {
                internalType: "bytes32",
                name: "hashValue",
                type: "bytes32",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bytes32",
                name: "hashType",
                type: "bytes32",
            },
        ],
        name: "hashTypeToSignersHash",
        outputs: [
            {
                internalType: "bytes32",
                name: "signersHash",
                type: "bytes32",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bytes32",
                name: "hashType",
                type: "bytes32",
            },
        ],
        name: "hashes",
        outputs: [
            {
                internalType: "bytes32",
                name: "hashValue",
                type: "bytes32",
            },
            {
                internalType: "uint256",
                name: "hashTimestamp",
                type: "uint256",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [],
        name: "owner",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bytes32",
                name: "hashType",
                type: "bytes32",
            },
            {
                internalType: "bytes32",
                name: "hashValue",
                type: "bytes32",
            },
            {
                internalType: "uint256",
                name: "hashTimestamp",
                type: "uint256",
            },
            {
                internalType: "bytes[]",
                name: "signatures",
                type: "bytes[]",
            },
        ],
        name: "registerHash",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [],
        name: "renounceOwnership",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bytes32",
                name: "hashType",
                type: "bytes32",
            },
            {
                internalType: "bytes32",
                name: "hashValue",
                type: "bytes32",
            },
        ],
        name: "setHash",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bytes32",
                name: "hashType",
                type: "bytes32",
            },
            {
                internalType: "address[]",
                name: "signers",
                type: "address[]",
            },
        ],
        name: "setSigners",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [],
        name: "signatureDelegationHashType",
        outputs: [
            {
                internalType: "bytes32",
                name: "",
                type: "bytes32",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "newOwner",
                type: "address",
            },
        ],
        name: "transferOwnership",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
];
class IHashRegistry__factory {
    static abi = _abi;
    static createInterface() {
        return new ethers_1.Interface(_abi);
    }
    static connect(address, runner) {
        return new ethers_1.Contract(address, _abi, runner);
    }
}
exports.IHashRegistry__factory = IHashRegistry__factory;
