// @api3/contracts/dist/typechain-types/factories/access/interfaces/IOwnableCallForwarder__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IOwnableCallForwarder__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [
            {
                internalType: "address",
                name: "forwardTarget",
                type: "address",
            },
            {
                internalType: "bytes",
                name: "forwardedCalldata",
                type: "bytes",
            },
        ],
        name: "forwardCall",
        outputs: [
            {
                internalType: "bytes",
                name: "returnedData",
                type: "bytes",
            },
        ],
        stateMutability: "payable",
        type: "function",
    },
];
class IOwnableCallForwarder__factory {
    static abi = _abi;
    static createInterface() {
        return new ethers_1.Interface(_abi);
    }
    static connect(address, runner) {
        return new ethers_1.Contract(address, _abi, runner);
    }
}
exports.IOwnableCallForwarder__factory = IOwnableCallForwarder__factory;
